examples/aggregator/client/aggregator.cc

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "asylo/util/logging.h"
+#include "examples/aggregator/proto/aggregator.grpc.pb.h"
+#include "examples/aggregator/proto/aggregator.pb.h"
+#include "include/grpcpp/grpcpp.h"
+#include "oak/client/application_client.h"
+
+ABSL_FLAG(std::string, address, "127.0.0.1:8080", "Address of the Oak application to connect to");
+
+using ::oak::examples::aggregator::Aggregator;
+using ::oak::examples::aggregator::Vector;
+
+void submit_sample(Aggregator::Stub* stub, std::vector<uint64_t> values) {
+  grpc::ClientContext context;
+  Vector request;
+  for (auto value : values) {
+    request.add_items(value);
+  }
+  google::protobuf::Empty response;
+  grpc::Status status = stub->SubmitSample(&context, request, &response);
+  if (!status.ok()) {
+    LOG(QFATAL) << "Could not submit sample: " << status.error_code() << ": "
+                << status.error_message();
+  }
+}
+
+void get_aggregation(Aggregator::Stub* stub) {
+  grpc::ClientContext context;
+  google::protobuf::Empty request;
+  Vector response;
+  grpc::Status status = stub->GetCurrentValue(&context, request, &response);
+  if (!status.ok()) {
+    LOG(WARNING) << "Could not get current value: " << status.error_code() << ": "
+                 << status.error_message();
+    return;
+  }
+
+  LOG(INFO) << "Aggregation:";
+  for (auto value : response.items()) {
+    LOG(INFO) << "- " << value;
+  }
+}
+
+int main(int argc, char** argv) {
+  absl::ParseCommandLine(argc, argv);
+
+  std::string address = absl::GetFlag(FLAGS_address);
+  LOG(INFO) << "Connecting to Oak Application: " << address;
+
+  oak::ApplicationClient::InitializeAssertionAuthorities();
+
+  auto stub = Aggregator::NewStub(oak::ApplicationClient::CreateChannel(address));
+
+  // Submit initial samples.
+  submit_sample(stub.get(), {0, 0, 10, 10, 0});
+  submit_sample(stub.get(), {10, 10, 0, 0, 10});
+
+  // Try to retrieve aggregation.
+  get_aggregation(stub.get());
```

Comment:
It seems that this client may as well check that the returned values are as expected, since the flow is very deterministic. Basically turn this into a test, that can be run and checks some assertions.

Reply:
(drive-by) ISTR elsewhere we used [`absl::optional`](https://github.com/abseil/abseil-cpp/blob/master/absl/types/optional.h) to avoid the need for C++17.